#! /bin/sh -e
## 09_use_gnu_source.dpatch by Joshua Kwan <joshk@triplehelix.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Define _GNU_SOURCE for RTLD_NEXT.

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1;;
esac

exit 0

--- a/sys/unix/unixres.c~	2003-10-06 22:02:16.000000000 -0700
+++ b/sys/unix/unixres.c	2003-10-06 22:02:33.000000000 -0700
@@ -21,6 +21,7 @@
 # if defined(LINUX)
 
 /* requires dynamic linking with libc */
+#define _GNU_SOURCE
 #include <dlfcn.h>
 
 static int
